kartothek/io_components/utils.py

```diff
@@ -157,14 +160,19 @@ def validate_partition_keys(
     return ds_factory, ds_metadata_version, partition_on
 
 
-_ARGS_TO_TYPE = {"partition_on": list, "delete_scope": list, "secondary_indices": list}
+_ARGS_TO_TYPE = {"partition_on": list, "delete_scope": list, "secondary_indices": set}
 
 
 def normalize_arg(arg_name, old_value):
     def _make_list(*_args):
         return list(_args)
 
-    type_to_normalize = {list: _make_list}
+    def _make_set(x):
+        if isinstance(x, (str, bytes)):
```

Comment:
This introduces different behavior for lists and sets. If we catch this edge case for strings/bytes we should do the same for lists. afaik, we're doing this for partition_on already but I'm not sure where and if this is consistently done

Reply:
Yes, the same happens for lists.
The thing is, any object you pass  (e.g. the tuple `(0,1)`) to the `_make_list` normalization function, the return value is a list with that object as the element, e.g. `[(0, 1)]`.
I wanted to avoid that in the `_make_set` normalization, so that lists are normalized to a proper set, to not cause a breaking change.
The underlying issue is that the normalization in `normalize_arg` is not strict nor really clear, as stated in #56 . It could probably use a re-write, but that would be out of scope for this PR.